source/base/mu_parser_internal.cc

```diff
@@ -155,53 +155,320 @@ namespace internal
       return uniform_distribution(rng);
     }
 
-    std::vector<std::string> function_names = {
-      // functions predefined by muparser
-      "sin",
-      "cos",
-      "tan",
-      "asin",
-      "acos",
-      "atan",
-      "sinh",
-      "cosh",
-      "tanh",
-      "asinh",
-      "acosh",
-      "atanh",
-      "atan2",
-      "log2",
-      "log10",
-      "log",
-      "ln",
-      "exp",
-      "sqrt",
-      "sign",
-      "rint",
-      "abs",
-      "min",
-      "max",
-      "sum",
-      "avg",
-      // functions we define ourselves above
-      "if",
-      "int",
-      "ceil",
-      "cot",
-      "csc",
-      "floor",
-      "sec",
-      "pow",
-      "erf",
-      "erfc",
-      "rand",
-      "rand_seed"};
+    std::vector<std::string>
+    get_function_names()
+    {
+      return {// functions predefined by muparser
+              "sin",
+              "cos",
+              "tan",
+              "asin",
+              "acos",
+              "atan",
+              "sinh",
+              "cosh",
+              "tanh",
+              "asinh",
+              "acosh",
+              "atanh",
+              "atan2",
+              "log2",
+              "log10",
+              "log",
+              "ln",
+              "exp",
+              "sqrt",
+              "sign",
+              "rint",
+              "abs",
+              "min",
+              "max",
+              "sum",
+              "avg",
+              // functions we define ourselves above
+              "if",
+              "int",
+              "ceil",
+              "cot",
+              "csc",
+              "floor",
+              "sec",
+              "pow",
+              "erf",
+              "erfc",
+              "rand",
+              "rand_seed"};
```

Comment:
Should this list be `static` and the function return a `const &` to it?

Reply:
We could do that, but we only call this when we set up the muParser objects so its not going to make a measurable performance difference. I would prefer not to add extra static objects unless we need to.